src/libraries/Microsoft.PowerFx.Connectors/Tabular/TableMetadata.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    internal sealed class TableMetadata : IExternalTableMetadata
+    {
+        private readonly IReadOnlyDictionary<string, ColumnMetadata> _columnsByName;
+
+        public TableMetadata(string name, string displayName, bool isReadOnly, IReadOnlyList<ColumnMetadata> columns, string parameterPkColumnName = "")
```

Comment:
Assuming this is coming directly from PA? If so, we may also want to move any tests they already may have related to this.

Reply:
This is a simplified version as they include NavigationMetadata
That said, I haven't found any specific test for this class